src/pudl_archiver/archivers/classes.py

```diff
@@ -364,6 +367,73 @@ def _check_dataset_size(
             required_for_run_success=self.fail_on_dataset_size_change,
         )
 
+    def _check_data_continuity(
+        self, new_datapackage: DataPackage
+    ) -> validate.DatasetSpecificValidation:
+        """Check that the archived data partitions are continuous and unique."""
+        success = True
+        note = None
+        partition_names = []
+        dataset_partitions = []
+
+        # Unpack partitions of a dataset.
+        for resource in new_datapackage.resources:
+            if not resource.parts:  # Skip resources without partitions
+                continue
+            partition_names += list(resource.parts)
+            dataset_partitions += list(resource.parts.values())[0]
+
+        partition_names = set(partition_names)
+
+        # Only perform this test if the part label is year quarter or year month
+        # Note that this currently only works if there is one set of partitions,
+        # and will fail if year_month and form are used to partition a dataset, e.g.
+        if any(partition in VALID_PARTITION_RANGES for partition in partition_names):
+            partition_to_test = [
+                partition
+                for partition in partition_names
+                if partition in VALID_PARTITION_RANGES
+            ][0]
+            interval = (
+                "QS"
+                if partition_to_test == "year_quarter"
+                else "MS"
+                if partition_to_test == "year_month"
+                else None
+            )
+
+            if interval:
```

Comment:
doesn't this if have the same functionality as the `if any(partition in VALID_PARTITION_RANGES for partition in partition_names):`? could we remove one of them? could `VALID_PARTITION_RANGES` be a dictionary of `part: freq`? so you don't have to to this nested if-ing?

Reply:
Agreed and done